src/element-style-observer.js

```diff
@@ -96,17 +97,40 @@ export default class ElementStyleObserver {
 	/**
 	 * Called the first time observe() is called to initialize the target.
 	 */
-	#init() {
+	#init () {
 		if (this.#initialized) {
 			return;
 		}
 
 		let firstTime = this.constructor.all.get(this.target).size === 1;
 		this.updateTransition({ firstTime });
 
+		// Initialize transition behavior for the main document
+		this.#setTransitionBehavior(this.target.ownerDocument);
+
+		// If the target is in a shadow root, initialize that too
```

Comment:
Is init the best place for this? What if the element is later moved to another shadow root or document? 

One way would be to cache the element's root node in a [`WeakRef`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakRef) and then do this both on init as well as on reparenting (which we know happened if  IntersectionObserver fires + the root node has changed)